### PR TITLE
Make sure glob does not consume recursive directories

### PR DESCRIPTION
**Pull request by tvoinarovskyi**
I know the fix is a bit hacky, but can't find a better way for now. Tried to do a recursive iteration, as it's done in storage but it did not perform very well, it needs some attention to do it right. What do you think of this fast fix for now?

This is the related function in gsutil:
https://github.com/GoogleCloudPlatform/gsutil/blob/master/gslib/wildcard_iterator.py
Quite a lot of edge cases as it seems.

### CODE REVIEW

**Comment by asvetlov** [comment]
Do you have a thought about how it can be implemented in an iterative manner without fetching everything in memory?
A number of objects in bucket is unlimited as far as I know.

**Comment by serhiy-storchaka** [comment]
What exactly problem it solves?

**Comment by tvoinarovskyi** [comment]
@serhiy-storchaka So the problem case is as follows:
```
neuro blob glob *.jpg
```
will return all subdirectories, while I would expect it to return only 1 folder deep.

**Comment by tvoinarovskyi** [comment]
@asvetlov Yes I do have thoughts, but will require to spend a good 2 days implementing, so I proposed this hack to handle the case for now.

**Comment by tvoinarovskyi** [comment]
Relates to https://github.com/neuromation/platform-client-python/pull/1445

File: neuromation/api/blob_storage.py [hunk 1]
```diff
@@ -217,6 +218,7 @@ def __init__(self, core: _Core, config: Config) -> None:
     async def glob_blobs(self, bucket_name: str, pattern: str) -> List[BlobListing]:
         pattern = pattern.lstrip("/")
         parts = pattern.split("/")
```
Comment (serhiy-storchaka) [inline neuromation/api/blob_storage.py]: Humm, the pattern can contain something like `[-/]`.

File: neuromation/api/blob_storage.py [hunk 2]
```diff
@@ -225,7 +227,40 @@ def __init__(self, core: _Core, config: Config) -> None:
...
+                for matcher, part in itertools.zip_longest(matchers, parts):
+                    if part is None or matcher is None:
+                        return False
+                    if not matcher(part):
+                        return False
+                return True
+
+        else:
+            matcher = "/".join(matcher_parts) + r"\Z"
+            match = re.compile(matcher).fullmatch
```
[the remark this answers is not in the document]
Comment (tvoinarovskyi) [reply neuromation/api/blob_storage.py:262]: Oh, yea...

File: neuromation/api/blob_storage.py (hunk 1 above)
Comment (tvoinarovskyi) [reply neuromation/api/blob_storage.py:220]: Then it would be an invalid URL.

Comment (serhiy-storchaka) [reply neuromation/api/blob_storage.py:220]: It is not a URL, it is a pattern. `foo[-/]bar` matches two paths: `foo-bar` and `foo/bar`.

Comment (tvoinarovskyi) [reply neuromation/api/blob_storage.py]: Is that a valid glob case?

**Pull request merged**